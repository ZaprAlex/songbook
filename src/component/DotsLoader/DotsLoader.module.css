.dot-loader {
    display: flex;
    align-items: center;
}

.dot-loader > div {
    width: 12px;
    height: 12px;

    background-color: #000000;
    border-radius: 100%;
    margin: 3px;
    -webkit-animation-fill-mode: both;
    animation-fill-mode: both;
    display: inline-block;
}

.dot-loader.white > div {
    background-color: #ffffff;
}

.dot-loader.black > div {
    background-color: #000000;
}

.dot-loader > div:nth-child(1) {
    -webkit-animation: scale 1.44s -0.24s infinite cubic-bezier(0, 0, 0.4, 1);
    animation: scale 1.44s -0.24s infinite cubic-bezier(0, 0, 0.4, 1);
}

.dot-loader > div:nth-child(2) {
    -webkit-animation: scale 1.44s -0.12s infinite cubic-bezier(0, 0, 0.4, 1);
    animation: scale 1.44s -0.12s infinite cubic-bezier(0, 0, 0.4, 1);
}

.dot-loader > div:nth-child(3) {
    -webkit-animation: scale 1.44s 0s infinite cubic-bezier(0, 0, 0.4, 1);
    animation: scale 1.44s 0s infinite cubic-bezier(0, 0, 0.4, 1);
}

@-webkit-keyframes scale {
    0% {
        -webkit-transform: scale(1);
        transform: scale(1);
        opacity: 1;
    }
    24% {
        -webkit-transform: scale(0.72);
        transform: scale(0.72);
        opacity: 0.3;
    }
    48% {
        -webkit-transform: scale(1);
        transform: scale(1);
        opacity: 1;
    }
}

@keyframes scale {
    0% {
        -webkit-transform: scale(1);
        transform: scale(1);
        opacity: 1;
    }
    24% {
        -webkit-transform: scale(0.72);
        transform: scale(0.72);
        opacity: 0.3;
    }
    48% {
        -webkit-transform: scale(1);
        transform: scale(1);
        opacity: 1;
    }
}
