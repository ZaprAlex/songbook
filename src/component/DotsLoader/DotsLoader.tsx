import React from 'react';
import cn from 'classnames';

import styles from './DotsLoader.module.css';

type Props = {
    color?: 'white' | 'black';
};

const DotsLoader: React.FC<Props> = ({ color = 'white' }) => (
    <div className={cn(styles.dotLoader, { [styles[color]]: true })}>
        <div />
        <div />
        <div />
    </div>
);

export default DotsLoader;
