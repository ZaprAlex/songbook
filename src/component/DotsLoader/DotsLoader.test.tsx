import React from 'react';
import renderer from 'react-test-renderer';

import DotsLoader from './DotsLoader';

describe('DotsLoader', () => {
    test('Snap w/o props', () => {
        const component = renderer.create(<DotsLoader />).toJSON();
        expect(component).toMatchSnapshot();
    });

    test('Snap with color prop', () => {
        const component = renderer.create(<DotsLoader color="black" />).toJSON();
        expect(component).toMatchSnapshot();
    });

    test('Snap with color props', () => {
        const component = renderer.create(<DotsLoader color="black" />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
