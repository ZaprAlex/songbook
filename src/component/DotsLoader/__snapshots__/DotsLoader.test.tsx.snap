// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotsLoader Snap w/o props 1`] = `
<div
  className="dotLoader white"
>
  <div />
  <div />
  <div />
</div>
`;

exports[`DotsLoader Snap with color prop 1`] = `
<div
  className="dotLoader black"
>
  <div />
  <div />
  <div />
</div>
`;

exports[`DotsLoader Snap with color props 1`] = `
<div
  className="dotLoader black"
>
  <div />
  <div />
  <div />
</div>
`;
